### ISSUE TITLE
Storage schema upgrade failed at v1 on update to V4.23

### ISSUE THREAD

**Issue by Elmardus** [authelia/authelia#1441]
On updating to the new release (using docker-compose) from V4.22 to 4.23, Authelia fails to initialize the SQL database with the following error:

```
time="2020-11-10T11:37:31+02:00" level=info msg="Logging severity set to debug"
time="2020-11-10T11:37:31+02:00" level=debug msg="Storage schema is being checked to verify it is up to date"
time="2020-11-10T11:37:31+02:00" level=debug msg="Storage schema is v0, latest is v1"
time="2020-11-10T11:37:31+02:00" level=fatal msg="Unable to initialize SQL database: storage schema upgrade failed at v1: Unable to /mysql_provider.go:77 NewMySQLProvidergithub.com/authelia/authelia/cmd/authelia/main.go:76               startServer\ngithub.com/aommand.go:943                       (*Command).ExecuteC\ngithub.com/spf13/cobra@v0.0.7/command.go:883                       (*Comman                                        goexit"
```

Reverting to `image: authelia/authelia:4.22` resolves this issue. I am using the latest mariadb docker-image, set up in configuration.yml as shown below:

```
storage:
  mysql:
    host: authelia-db
    port: 3306
    database: authelia
    username: authelia
    password: redacted
```

Did I configure my database incorrectly? Or is this caused by a recent change?

**Comment by james-d-elliott** [comment]
We added one table. What MariaDB version did you use? I'm sure we tested that upgrade, wondering if it's something specific to the version or your setup.

Does the Authelia user have create table perms on that DB?

**Comment by Elmardus** [comment]
I am on the lastest mariadb docker image, which is _1:10.5.7+maria~focal_. 
To be honest I am not sure about the permissions on that DB, I haven't configured anything in the mariadb except the following in the docker-compose file:

```
  authelia:
    image: authelia/authelia
    container_name: authelia
    volumes:
      - /home/Authelia:/config
    expose:
      - 9091
    restart: unless-stopped
    depends_on:
      - authelia-db
      - authelia-redis

  authelia-db:
    image: mariadb
    container_name: authelia-db
    restart: unless-stopped
    expose:
      - 3306      
    volumes:
      - /home/Authelia/db:/var/lib/mysql
    environment:
      - MYSQL_ROOT_PASSWORD=redacted
      - MYSQL_PASSWORD=redacted
      - MYSQL_DATABASE=authelia
      - MYSQL_USER=authelia
```

The following tables exist now (I reverted to 4.22)
```
root@b2d6ebee6668:/# mysqlshow authelia -p               
Database: authelia
+------------------------------+
|            Tables            |
+------------------------------+
| authentication_logs          |
| config                       |
| identity_verification_tokens |
| totp_secrets                 |
| u2f_devices                  |
| user_preferences             |
+------------------------------+
```

**Comment by james-d-elliott** [comment]
So you have the new table (config), so it failed on the index or replace into statement. I'll see if I can replicate it. If you get time I'd appreciate a `SELECT * FROM config`.

**Comment by Elmardus** [comment]
This is the result, I attached all commands just to be sure:
```
root@b2d6ebee6668:/# mysql -uauthelia -p authelia
Enter password: 
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 16
Server version: 10.5.7-MariaDB-1:10.5.7+maria~focal mariadb.org binary distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [authelia]> SELECT * FROM config;
+----------+----------+-------+
| category | key_name | value |
+----------+----------+-------+
| schema   | version  | 1     |
+----------+----------+-------+
1 row in set (0.001 sec)
```

**Comment by Elmardus** [comment]
I've managed to workaround the issue by using the linuxserver/mariadb container instead of the official mariadb image. However I wasn't able to find the root cause why the official mariadb did not work. Even on a clean installation (deleted mariadb container & db folder) the following error occured:

```
time="2021-02-20T00:57:46+01:00" level=info msg="Logging severity set to debug"
time="2021-02-20T00:57:46+01:00" level=debug msg="Storage schema is being checked to verify it is up to date"
time="2021-02-20T00:57:46+01:00" level=debug msg="Storage schema is v0, latest is v1"
time="2021-02-20T00:57:46+01:00" level=fatal msg="Unable to initialize SQL database: storage schema upgrade failed at v1: Unable to create table authentication_logs: Error 1050: Table 'authentication_logs' already exists" stack="github.com/authelia/authelia/internal/storage/mysql_provider.go:77 NewMySQLProvider\ngithub.com/authelia/authelia/cmd/authelia/main.go:80               startServer\ngithub.com/authelia/authelia/cmd/authelia/main.go:137              main.func1\ngithub.com/spf13/cobra@v1.1.1/command.go:854                       (*Command).execute\ngithub.com/spf13/cobra@v1.1.1/command.go:958                       (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.1.1/command.go:895                       (*Command).Execute\ngithub.com/authelia/authelia/cmd/authelia/main.go:154              main\nruntime/proc.go:204                                                main\nruntime/asm_amd64.s:1374                                           goexit"
```

**Comment by kailaru** [comment]
I'm encountering this error with version v.32.2.
```
time="2021-11-03T15:24:29Z" level=info msg="Authelia v4.32.2 is starting"
time="2021-11-03T15:24:29Z" level=info msg="Log severity set to info"
time="2021-11-03T15:24:29Z" level=fatal msg="Unable to initialize SQL database: storage schema upgrade failed at v1: unable to create table authentication_logs: ERROR: relation \"authentication_logs\" already exists (SQLSTATE 42P07)" stack="github.com/authelia/authelia/v4/internal/storage/postgres_provider.go:86 NewPostgreSQLProvider\ngithub.com/authelia/authelia/v4/internal/commands/root.go:99             getProviders\ngithub.com/authelia/authelia/v4/internal/commands/root.go:68             cmdRootRun\ngithub.com/spf13/cobra@v1.2.1/command.go:860                             (*Command).execute\ngithub.com/spf13/cobra@v1.2.1/command.go:974                             (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.2.1/command.go:902                             main\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:11                  main\nruntime/proc.go:255                                                      main\nruntime/asm_amd64.s:1581                                                 goexit"
```
I tried wiping the database but it didn't change the error.
Output of `SELECT * from config;`
```
authelia=> SELECT * FROM config;
 category | key_name | value 
----------+----------+-------
 schema   | version  | 1
(1 row)
```

**Comment by james-d-elliott** [comment]
That's pretty strange if it's from an empty DB. What version of postgres?

**Comment by james-d-elliott** [comment]
I believe this will be fixed in 4.33.0 (when released). I can't see why it originally happened, but we're moving to a much more robust migration method. If you're still having this issue post 4.33 then I'll get you to pop on Matrix/Discord and I'll manually help you fix the issues.

**Comment by kailaru** [comment]
@james-d-elliott Thanks, I just upgraded to 4.33.2 today and it solved the bug. I apologize for the slow response, this fell of my radar for a bit.
